rules/naming/tests/ValueObjectFactory/PropertyRenameFactory/PropertyRenameFactoryTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Naming\Tests\ValueObjectFactory\PropertyRenameFactory;
+
+use Iterator;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassLike;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\Stmt\PropertyProperty;
+use Rector\Core\HttpKernel\RectorKernel;
+use Rector\Core\PhpParser\Node\BetterNodeFinder;
+use Rector\FileSystemRector\Parser\FileInfoParser;
+use Rector\Naming\ValueObject\PropertyRename;
+use Rector\Naming\ValueObjectFactory\PropertyRenameFactory;
+use Symplify\PackageBuilder\Tests\AbstractKernelTestCase;
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+class PropertyRenameFactoryTest extends AbstractKernelTestCase
+{
+    /**
+     * @var PropertyRenameFactory
+     */
+    private $propertyRenameFactory;
+
+    /**
+     * @var FileInfoParser
+     */
+    private $fileInfoParser;
+
+    /**
+     * @var BetterNodeFinder
+     */
+    private $betterNodeFinder;
+
+    public function __construct(?string $name = null, array $data = [], $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
+
+        $this->bootKernel(RectorKernel::class);
+
+        $this->propertyRenameFactory = self::$container->get(PropertyRenameFactory::class);
+        $this->fileInfoParser = self::$container->get(FileInfoParser::class);
+        $this->betterNodeFinder = self::$container->get(BetterNodeFinder::class);
+    }
+
+    /**
+     * @dataProvider createDataProvider
+     */
+    public function testCreate(Property $property, array $expectedPropertyData): void
+    {
+        /** @var PropertyRename $actualPropertyRename */
+        $actualPropertyRename = $this->propertyRenameFactory->create($property);
+
+        $this->assertSame($property, $actualPropertyRename->getProperty());
+        $this->assertSame($expectedPropertyData['expectedName'], $actualPropertyRename->getExpectedName());
+        $this->assertSame($expectedPropertyData['currentName'], $actualPropertyRename->getCurrentName());
+        $this->assertInstanceOf(PropertyProperty::class, $actualPropertyRename->getPropertyProperty());
+        $this->assertInstanceOf(ClassLike::class, $actualPropertyRename->getClassLike());
```

Comment:
To be honest, I'm not sure how it could be tested better. I'm opened to propositions :)